morphia/src/main/java/org/mongodb/morphia/Morphia.java

```diff
@@ -255,7 +266,7 @@ public synchronized Morphia mapPackage(final String packageName, final boolean i
      * @return this
      */
     public Morphia mapPackageFromClass(final Class clazz) {
-        return mapPackage(clazz.getPackage().getName(), false);
+        return mapPackage(clazz.getPackage().getName(), false, true);
```

Comment:
While mapping a class (instead of package), shouldn't the subpackage mapping always be ignored?
Or do you mean that I should remove the mapSubPackage(...) method and include the functionality in the existing mapPackage(...)